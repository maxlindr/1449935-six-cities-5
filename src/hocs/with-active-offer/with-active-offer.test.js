import React from 'react';
import renderer from 'react-test-renderer';
import PropTypes from 'prop-types';
import withActiveOffer from './with-active-offer';
import offerMocks from '../../../__mocks__/offer-mocks';
import offerPropTypes from '../../prop-types/offer-prop-types';

const MockComponent = () => <div>Mock Component</div>;

MockComponent.propTypes = {
  activeOffer: offerPropTypes,
  onActivate: PropTypes.func.isRequired,
  onDeactivate: PropTypes.func.isRequired,
};

const MockComponentWrapped = withActiveOffer(MockComponent);

test(`withActiveOffer должен корректно отображаться`, () => {
  const tree = renderer.create((
    <MockComponentWrapped offers={offerMocks} />
  )).toJSON();

  expect(tree).toMatchSnapshot();
});
