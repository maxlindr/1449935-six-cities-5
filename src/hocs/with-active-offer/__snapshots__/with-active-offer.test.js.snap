// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withActiveOffer должен корректно отображаться 1`] = `
<div>
  Mock Component
</div>
`;
